prow/plugins/config.go

```diff
@@ -982,3 +983,135 @@ func (pluginConfig *ProjectConfig) GetOrgColumnMap(org string) map[string]string
 	}
 	return nil
 }
+
+// Bugzilla holds options for checking Bugzilla bugs in a defaulting hierarchy.
+type Bugzilla struct {
+	// Default settings mapped by branch in any repo in any org.
+	// The `*` wildcard will apply to all branches.
+	Default map[string]BugzillaBranchOptions `json:"default"`
+	// Options for specific orgs. The `*` wildcard will apply to all orgs.
+	Orgs map[string]BugzillaOrgOptions `json:"orgs"`
+}
+
+// BugzillaOrgOptions holds options for checking Bugzilla bugs for an org.
+type BugzillaOrgOptions struct {
+	// Default settings mapped by branch in any repo in this org.
+	// The `*` wildcard will apply to all branches.
+	Default map[string]BugzillaBranchOptions `json:"default"`
+	// Options for specific repos. The `*` wildcard will apply to all repos.
+	Repos map[string]BugzillaRepoOptions `json:"repos"`
+}
+
+// BugzillaRepoOptions holds options for checking Bugzilla bugs for a repo.
+type BugzillaRepoOptions struct {
+	// Options for specific branches in this repo.
+	// The `*` wildcard will apply to all branches.
+	Branches map[string]BugzillaBranchOptions `json:"branches"`
+}
+
+// BugzillaBranchOptions describes how to check if a Bugzilla bug is valid or not.
+type BugzillaBranchOptions struct {
+	IsOpen        *bool   `json:"is_open,omitempty"`
+	TargetRelease *string `json:"target_release,omitempty"`
+}
+
+func (o BugzillaBranchOptions) matches(other BugzillaBranchOptions) bool {
```

Comment:
nit: Should we just use `reflect.DeepEqual` so that this doesn't have to be maintained as new option fields are added?

Reply:
I didn't want to have us running `reflect` code since it's really slow and you could end up doing this a lot with nested config